roles/addons/nic_nmcli/tasks/main.yml

```diff
@@ -41,15 +41,66 @@
 #    - ansible_facts.distribution_major_version == "18"
 #    - ansible_facts.os_family == "Ubuntu"
 
-# Note from JK
-# We could add so much more features here, like VLAN, LACP, MTU, etc. Easy to add.
-- name: nmcli █ Set NIC configuration
+- name: nmcli █ Set NICs configuration
   nmcli:
-    conn_name: "{{ item.name | string }}"
-    type: "{% if item.type is defined and item.type is not none %}{{ item.type }}{% else %}ethernet{% endif %}"
-    ip4: "{{ item.ip4 }}/{{ networks[item.network].prefix }}"
-    state: present
-    gw4: "{% if networks[item.network].gateway is defined and networks[item.network].gateway is not none %}{{ networks[item.network].gateway }}{% endif %}"
+    # Tuned
+    conn_name: "{{ item.conn_name | default(item.interface) | string }}"
+    ifname: "{{ item.ifname | default(item.physical_device) | default(omit) }}"
+    ip4: "{{ (item.ip4 is defined) | ternary( ('/' in (item.ip4|default('None')) ) | ternary( (item.ip4|default('None')), ( (item.ip4|default('None')) | string) + '/' + (networks[item.network]['prefix4'] | default(networks[item.network]['prefix']) | string)), omit) }}"
+    # Equivalent to
+    # {% if item.ip4 is defined %}
+    #   {% if '/' in item.ip4 %}\
+    #     {{ item.ip4 }}
+    #   {% else %}\
+    #     {{ item.ip4 }}/{{ networks[item.network].prefix }}
+    #   {% endif %}
+    # {% else %}
+    #    omit
+    # {% endif %}
+    # Allows to provide a simple ip4, or an ip4/prefix
+    # Also support old prefix format (to be replaced by prefix4)
+    ip6: "{{ (item.ip6 is defined) | ternary( ('::' in (item.ip6|default('None')) ) | ternary( (item.ip6|default('None')), ( (item.ip6|default('None')) | string) + '::' + ((networks[item.network].prefix6|default('None')) |string)), omit) }}"
```

Comment:
Is ipv6 fully supported by the stack ?

Reply:
Not really. I was thinking of possible login nodes that would have 2 network, one internal in ipv4 and one external in ipv6.
Not sure it can really happen in fact...

Reply:
I will keep ipv6, but only as basic option, for external interfaces. Which means I will remove all 4/6 related elements. Should be done soon.